subprojects/internal-performance-testing/src/main/groovy/org/gradle/performance/fixture/AndroidTestProject.groovy

```diff
@@ -87,8 +88,12 @@ class AndroidTestProject implements TestProject {
         configureForAgpVersion(runner, AGP_VERSIONS.latest)
     }
 
-    private static void configureForAgpVersion(CrossVersionPerformanceTestRunner runner, String agpVersion) {
+    static void useJavaVersion(CrossVersionPerformanceTestRunner runner, JavaVersion javaVersion) {
+        def buildJavaHome = AvailableJavaHomes.getJdk(javaVersion).javaHome
+        runner.addBuildMutator { invocation -> new JavaVersionMutator(invocation, javaVersion, buildJavaHome) }
+    }
 
+    private static void configureForAgpVersion(CrossVersionPerformanceTestRunner runner, String agpVersion) {
         runner.args.add("-DagpVersion=${agpVersion}")
```

Comment:
💭 AGP and KGP versions

For SantaTracker we pass AGP and Kotlin versions `-DagpVersion=8.1.1 -DkotlinVersion=1.9.10`
NowInAndroid doesn't take these sysprops into account.
It uses a version catalog https://github.com/android/nowinandroid/blob/main/gradle/libs.versions.toml
We could have a build mutator that changes the version catalog to set the AGP and Kotlin version we want.
This can happen post merge of the current branch, it's already good just to have it running.